### PR TITLE
Styles adjustments, Http client, json-server database, date-based viewing, date switching.

### PR DESCRIPTION
**Pull request by Foxar**

### CODE REVIEW

File: src/app/add-task-dialog/add-task-dialog.component.scss [hunk 1]
```diff
@@ -0,0 +1,19 @@
+
+:host{
+    align-items: flex-start;
+    
+    .newTaskDialog{
```
Comment (h4570) [inline src/app/add-task-dialog/add-task-dialog.component.scss]: `kebab-case` is the most popular naming convention for css

File: src/app/add-task-dialog/add-task-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+@Component({
+  selector: 'app-add-task-dialog',
+  templateUrl: './add-task-dialog.component.html',
+  styleUrls: ['./add-task-dialog.component.scss']
+})
+export class AddTaskDialogComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
```
Comment (h4570) [inline src/app/add-task-dialog/add-task-dialog.component.ts]: If is not used, you can remove it (with `implements OnInit` also)

File: src/app/date-header/date-header.component.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { Component, OnInit, OnChanges, Input, Output, EventEmitter } from '@angular/core';
+
+@Component({
+  selector: 'app-date-header',
+  templateUrl: './date-header.component.html',
+  styleUrls: ['./date-header.component.scss']
+})
+export class DateHeaderComponent implements OnInit, OnChanges {
```
Comment (h4570) [inline src/app/date-header/date-header.component.ts]: `OnInit` and `OnChanges` can be deleted i think

File: src/app/services/task.service.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import {HttpClient, HttpHeaders} from '@angular/common/http';
+import {Task} from '../task';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TaskService {
+  private tasksEndpoint = 'http://localhost:3000/tasks';
```
Comment (h4570) [inline src/app/services/task.service.ts]: Move this url to `environment.ts`

File: src/app/task.ts [hunk 5]
```diff
@@ -1,10 +1,8 @@
 export interface Task {
     id: number;
-    title: string;
-    description: string;
-    dateAdded: [number,number,number]   //Day,Month,Year
-    deadline?: [number,number,number]   //Same as above
-    
+    content: string;
+    dateAdded: [number,number,number];   //Day,Month,Year
```
Comment (h4570) [inline src/app/task.ts:4]: This is too magic I think, my proposition:
```
dateAdded: { day: number, month: number, year: number }
```

**Comment by Foxar** [comment]
Think I applied all of those, thanks!

**Pull request merged**